pkgs/top-level/python-packages.nix

```
@@ -4044,6 +4046,10 @@ in {
 
   pyspread = callPackage ../development/python-modules/pyspread { };
 
+  pysparse = callPackage ../development/python-modules/pysparse {
+    blas = pkgs.openblas;
```

not familiar with this package, but i would at the very least keep the aliasing consistent
```suggestion
    openblas = pkgs.openblas;
```

---

I mentioned this in a previous comment. What if we would like to use mkl in the future? blas is more consistent. (also atlas and blas are options as well)

---

you can have additional flags such as:

```
, pkgs
, withBlas ?  false
, withMKL ? false
....
```

and then later on:
```
buildInputs = [
...
] ++ lib.optional withBlas pkgs.openblas
++ lib.optional withMKL pkgs.mkl;
```

but, you can set them to be true when calling the package
```
fipy = callPackage .... { withMKL = true; }
```

really just depends on how to you want to structure injecting those options.

---

https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/python-packages.nix#L3817

---

This issue is that you cannot have withBlas = true and with MKL = true only one

---

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/tensorflow/default.nix
is a good example as it conditionally enables CudaSupport. And does an "intelligent" look-up of what instruction sets are available for a given architecture.

I'm not sure if you're trying to set a framework to use, or choose between many.